src/common.js

```diff
@@ -46,11 +52,26 @@ function error(msg, _continue) {
 }
 exports.error = error;
 
-// In the future, when Proxies are default, we can add methods like `.to()` to primitive strings.
-// For now, this is a dummy function to bookmark places we need such strings
-function ShellString(str) {
-  return str;
-}
+//@
+//@ ### ShellString(str)
+//@
+//@ Examples:
+//@
+//@ ```
+//@ var foo = ShellString('hello world');
+//@ ```
+//@
+//@ Turns a regular string into a string-like object similar to what each
+//@ command returns. This has special methods, like `.to()` and `.toEnd()`
+var ShellString = function (str, stderr) {
+  var that = new String(str);
+  that.to = wrap('to', _to, {idx: 1});
+  that.toEnd = wrap('toEnd', _toEnd, {idx: 1});
+  that.stdout = str;
+  that.stderr = typeof stderr !== 'undefined' ? stderr : state.error;
```

Comment:
The rationale was for commands like `sed()`, which can sometimes have `stderr` output, and still return a value (imagine one file exists, and one file doesn't exist).

One alternative would be to not default it, but to have each call to `ShellString()` explicitly have a `stderr` argument if it needs it.